server/admin/MatchManagement.js

```diff
@@ -0,0 +1,145 @@
+				if (g.getBlueSpy() == "") {
+					g.setBlueSpy(userID);
+					mess = "You are the Blue Spy Master";
+				}
+			} else if (position == "RS") {
+				if (g.getRedSpy() == "") {
+					g.setRedSpy(userID);
+					mess = "You are the Red Spy Master";
+				}
+			} else if (position == "RF"){
+				if (g.getRedField() == "") {
+					g.setRedField(userID);
+					mess = "You are the Red Field Agent";
+				}
+			}
+		} 
+		console.log(mess);
+		let retVal = {info: this.matchInfo(matchID), message: mess}
+		console.log("another");
+		return retVal;
+	},
+
+	//End the match.
+	endMatch: function (matchID) {
+		//Store in db and remove from maps.
+		let mess = matchesByID.delete(matchID);
+		console.log(mess);
+		return {message: "Match deleted"};
+	},
+
+	//Remove the player from the match.
+	leaveMatch: function (matchID, userID) {
+		let g = getGame(matchID);
+		if (g == undefined || g == null) return {info: "", message: "Match not found"};
+		if (g.getBlueField() == userID) {
+			g.setBlueField("");
+		} else if (g.getBlueSpy() == userID) {
+			g.setBlueSpy("");
+		} else if (g.getRedField() == userID) {
+			g.setRedField("");
+		} else if (g.getRedSpy() == userID) {
+			g.setRedSpy("");
+		}
+		console.log(mess);
+		return {message: "Left Match"};
+	},
+
+	//Reset the match.
+	resetMatch: function (matchID) {
+		let g = getGame(matchID);
+		if (g == undefined || g == null) return {info: "", message: "Match not found"};
+		let mess = g.resetMatch();
+		console.log(mess);
+		return {info: this.matchInfo(matchID), message: mess};
+	},
+
+	//Spy turn.
+	spyCommand: function (matchID, userID, numGuesses, word) {
+		let g = getGame(matchID);
+		if (g == undefined || g == null) return {info: "", message: "Match not found"};
+		let mess = "Move failed";
+		if ((userID == g.getBlueSpy() && g.getState()==gameState.BLUE_SPY) || 
+		(userID == g.getRedSpy() && g.getState()==gameState.RED_SPY)) {
+			mess = g.nextSpyHint(numGuesses, word);
+		}
+		return this.matchInfo(matchID);
+	},
+
+	//Field agent turn.
+	fieldGuess: function (matchID, userID, guess) {
+		let g = getGame(matchID);
+		if (g == undefined || g == null) return {info: "", message: "Match not found"};
+		let mess = "";
+		if ((userID == g.getBlueField() && g.getState()==gameState.BLUE_FIELD) || 
+		(userID == g.getRedField() && g.getState()==gameState.RED_FIELD)) {
+			mess = g.nextWordGuess(guess);
+		}
+		console.log(mess);
+		return {info: this.matchInfo(matchID), message: mess};
+	},
+
+	//End turn of field agent.
+	endTurn: function(matchID, userID) {
+		let g = getGame(matchID);
+		if (g == undefined || g == null) return {info: "", message: "Match not found"};
+		let mess = "";
+		if ((userID == g.getBlueField() && g.getState()==gameState.BLUE_FIELD) || 
+		(userID == g.getRedField() && g.getState()==gameState.RED_FIELD)) {
+			console.log("Calling end turn in game");
+			mess = g.endTurn();
+		}
+		console.log(mess);
+		return {info: this.matchInfo(matchID), message: mess};
+	}
+}
+

```

Comment:
Refactor this as a proper es6 class, like you did with the Game Engine. Use more descriptive names for varaibles, like "game" instead of "g"

Reply:
If you are referring to the MatchManager, I did it that way to make it a singleton. If you are referring to the return statement, it is supposed to be a json object anyway.

Reply:
Yeah I meant MatchManager. It would still be a singleton as long as you export the instance, e.g: ` module.exports = new GameMatch();`